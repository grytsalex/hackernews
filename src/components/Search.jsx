import React from "react";

export const Search = ({ value, onChange, onSubmit, children }) => (
  <form onSubmit={onChange}>
    <input type="text" value={value} onChange={onChange} />
    <button type="submit" onClick={onSubmit}>{children}</button>
  </form>
);
