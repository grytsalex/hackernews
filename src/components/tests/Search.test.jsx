import React from 'react';
import renderer from 'react-test-renderer'

import { Search } from '../Search';

describe('Search', () => {
    test('should have snapshot', () => {
        const component = renderer.create(<Search>Search...</Search>);

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})