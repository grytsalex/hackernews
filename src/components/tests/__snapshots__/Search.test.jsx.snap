// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should have snapshot 1`] = `
<form>
  <input
    type="text"
  />
  <button
    type="submit"
  >
    Search...
  </button>
</form>
`;
